### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by SekoViper**
# Summary
## Implement the following

- Create class Person with the following:

Instance vars: @id, @name, and @age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @age.
Setters for @name and @age.
Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Create class Student with the following:

Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯\(ツ)/¯".

- Create class Teacher with the following:

Inherits from Person.
Constructor extends parent's constructor by adding @specialization and a parameter for it.
Override can_use_services? so it always returns true

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    age > 18
```
Comment (tugsem) [inline person.rb]: - [x] Kindly fix this condition to be true if age is greater or equal to 18 as in the requirements.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Person
+  attr_accessor :name, :age, :parent_permission
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
```
Comment (tugsem) [inline person.rb:6]: - [ ] Kindly implement getter for the id variable as stated in the requirements: 
image

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class Student < Person
+  attr_accessor :classroom, :age, :name
+
+  def initialize(age, classroom = 'class one', name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
+
+  def play_hooky()
```
Comment (tugsem) [inline student.rb]: - [ ] Kindly remove empty parenthesis after this function name as in ruby, we don't use them if there are no arguments.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hi @SekoViper, 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️gitflow
✔️linters
✔️readme is professional
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (SekoViper) [reply person.rb]: Thanks. But the condition will return true if the the method is called.

File: person.rb (hunk 2 above)
Comment (mckent05) [reply person.rb:6]: - Please, just like the last reviewer mentioned implement getters for `@id`. Please fix this as it is part of the project requirements. 🙏

File: person.rb (hunk 1 above)
Comment (mckent05) [reply person.rb]: - Please fix the condition, according to the project requirements, it should check if the age is greater or equal to `18`

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    age > 18
+  end
+
+  def can_use_services?
+    of_age? && parent_permission
```
Comment (mckent05) [inline person.rb]: **OPTIONAL** - According to the project requirements this function should `return true if person is of age or if they have permission from parents`

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    true
+  end
+end
```
Comment (mckent05) [inline teacher.rb]: **OPTIONAL** - According to the project requirements, this class should Inherit from the Person class. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @SekoViper, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @SekoViper 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- The classes to represent students and teachers have been implemented 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**